### ISSUE TITLE
Diverse output images?

### ISSUE THREAD

**Issue by doantientai** [NVIDIA/pix2pixHD#60]
In the paper, it is written that by training the network E we are able to generate diverse images from given inputs. I have trained the network with **--instance_feat** and **--n_clusters 10**. However, when I run the test, I gave me only one synthetic image for each input sample. I think I am missing something. Please help!

**Comment by tcwang0509** [comment]
**--n_clusters 10** means it will randomly pick one from the 10 possible clusters. If you want to generate all 10 outputs, you will need to write a for loop.

**Comment by doantientai** [comment]
Thank you @tcwang0509 , I will try doing that. By the way, it seems not possible to train with --no_instance but still having diverse output, is it? I just want to make sure if I understand the paper correctly.

**Comment by naren142** [comment]
Hi @doantientai as you said you have tested the your trained model, may I know how are the results of your model? I am training my model with human images.Is the image generated by testing the model different from the training images?And if I do it for humans, will I get a different human image?

**Comment by doantientai** [comment]
@naren142 Sorry I didn't understand your question. As the purpose of Pix2pixHD is to transform images, it won't generate images from scratch like basic GANs do. So make sure you have paired samples to train the network.

In my case it was medical images. I use the network to transform 3d projections to realistic images. From what I see, the model always works much better on the training dataset, which is normal. However, the performance on the test set is somehow acceptable in my case.

**Comment by naren142** [comment]
So,if I train it on human image and corresponding mask(label) and give new mask as input what will it generate?

**Comment by doantientai** [comment]
@naren142 Yes in that case it will outputs colored human images. With the parameter --num_style you can chose to generate how many output from 1 input.

**Comment by naren142** [comment]
@doantientai Thanks. Is the generated image same as the training images or will it generate a new image?

**Comment by doantientai** [comment]
@naren142 The outputs on test set are similar to ones from training set in general, but they are not (and should not be) exactly the same to any certain sample from training set. You can take a look on images on the paper https://arxiv.org/abs/1804.04732

**Comment by daeunni** [comment]
@doantientai  Hi, where is an **--num_style**parameter in this code?
